package/MDAnalysis/analysis/base.py

```diff
@@ -179,13 +179,15 @@ def run(self, start=None, stop=None, step=None, verbose=None):
         # if verbose unchanged, use class default
         verbose = getattr(self, '_verbose',
                           False) if verbose is None else verbose
+        kwargs['verbose'] = verbose
 
         self._setup_frames(self._trajectory, start, stop, step)
         logger.info("Starting preparation")
         self._prepare()
+
         for i, ts in enumerate(ProgressBar(
                 self._trajectory[self.start:self.stop:self.step],
-                verbose=verbose)):
+                **kwargs)):
```

Comment:
May I ask why you didn't simply `verbose=verbose, **kwargs`? That way you don't have to modify the kwargs dictionary.

Reply:
Hey, I tried that, and yes, I won't have to modify the `kwargs` dictionary in the run function of base.py file, but the `kwargs` dictionary in the `__init__` function of log.py file is expecting verbose to be a key in its `kwargs` dictionary, but in the suggested way, verbose will be a named parameter, and not a key in the `kwargs` dictionary of the `__init__` function.
So, I would have to instead do the modifications over there.
This modification seemed cleaner to me, thus, I did it this way, but if the suggested way (`verbose=verbose, **kwargs`) is preferred, I will do it that way instead.

Reply:
Also, please guide me on creating the tests. From what I understand, I need to create a function at the end of the testsuite/MDAnalysisTests/analysis/test_base.py file, that calls the ProgressBar, with some kwargs, which will in turn be passed to tqdm, and consequently, affect the position of the ProgressBars.
But, I can't figure out how and what set of methods to call to do it, and how to verify whether my function is behaving as expected.
Thanks

Reply:
Have a look at the [tqdm documentation](https://github.com/tqdm/tqdm#documentation) for possible parameters you could pass through, and the existing ProgressBar tests in MDAnalysis for suggestions on how you could test for progress bar appearance. We also have a [brief guide](https://userguide.mdanalysis.org/stable/testing.html#writing-new-tests) for writing tests in the user guide.

Reply:
> Hey, I tried that, and yes, I won't have to modify the `kwargs` dictionary in the run function of base.py file, but the `kwargs` dictionary in the `__init__` function of log.py file is expecting verbose to be a key in its `kwargs` dictionary, but in the suggested way, verbose will be a named parameter, and not a key in the `kwargs` dictionary of the `__init__` function.
> So, I would have to instead do the modifications over there.

I don't think I understand what you mean here. Could you show me your suggested changes? If you just pass in `verbose=verbose, **kwargs`, the log.ProgressBar class will consider both of them to be in `kwargs` due to [dictionary unpacking](https://docs.python.org/3/tutorial/controlflow.html#keyword-arguments)